recipes/views.py

```diff
@@ -0,0 +1,63 @@
+from django.urls import reverse_lazy
+from django.views.generic import (CreateView, DetailView, DeleteView,
+                                  ListView, UpdateView)
+
+from recipes.models import Recipe
+
+
+class RecipeCreateView(CreateView):
+    """
+
+    View to create a recipe
+    """
+    model = Recipe
+    fields = ['user', 'title', 'description', 'servings', 'ingredients', 'preparation_time',
+              'cook_time', 'difficulty']
+
+    def get_success_url(self):
+        return reverse_lazy('detail-single-recipe', kwargs={'pk': self.object.id})
+
+
+class RecipeListView(ListView):
+    """
+
+    View that lists all recipes OR filters by Ingredient
+    """
+    model = Recipe
+    context_object_name = 'recipe_list'
+    ordering = ['-created_at']
```

Comment:
this can be set on the model too, if it always makes sense to order these the same, regardless of the view